crest/Assets/Crest/Crest-Examples/BoatDev/Scripts/BoatAlignNormal.cs

```diff
@@ -72,6 +68,11 @@ void Start()
             enabled = false;
             return;
         }
+
+        if (!_boatControl)
+        {
+            _boatControl = GetComponent<BoatControl>();
```

Comment:
Do we want this? Maybe it would be better to require the field to be assigned-to explicitly and then print an error and disable this script if this problem is encountered? (I personally like to keep things simple, and having one way of initialising as opposed to two reflects that, but please feel free to disagree on this point and don't let this be a merge-blocker).

Reply:
Good point. I think allowing null here will be useful. For example, perhaps using boat probes for a broken vessel. A null component could be created for that use case though. So maybe we could warn on validation instead?

Reply:
Or should we automatically attach a player control if no control is present?

Reply:
Maybe its a sensible default that can head off support requests.

Reply:
We removed that functionality for the register input components (auto assign material). Also, the control classes live in _Crest-Examples_ which users might opt not to import. I created a proposal with PR #497 that might help address this problem better.

Reply:
True. thinking it through, i think it wasnt obvious to me that theres a good default choice of the ocean input. i guess it comes down to how we feel about the default behaviour being to apply player controls. i'll defer to your judgement

Reply:
I have changed this to use validation instead of getting a component.